opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -444,5 +450,11 @@ def add_span_processor(self, span_processor: SpanProcessor) -> None:
         # thread safe
         self._active_span_processor.add_span_processor(span_processor)
 
+    def shutdown(self):
+        """Shutdown the span processors added to the tracer."""
+        self._active_span_processor.shutdown()
+        if hasattr(self, "_atexit_handler"):
```

Comment:
I'd prefer not using `hasattr` and instead assign `None` in `__init__` if we don't register a handler.